java/arcs/core/host/AbstractArcHost.kt

```diff
@@ -10,43 +10,239 @@
+        }
+
+        // Can't restart a deleted arc
+        if (context.arcState == ArcState.Deleted) {
+            return
+        }
+
+        for (particleSpec in partition.particles) {
+            context.particles[particleSpec] = startParticle(particleSpec, context)
+        }
+
+        performLifecycleForContext(context)
+        maybeRequestResurrection(context)
+
+        context.arcState = ArcState.Running
+        updateArcHostContext(partition.arcId, context)
+    }
+
+    suspend fun startParticle(spec: Plan.Particle, context: ArcHostContext): ParticleContext {
+        val particle = instantiateParticle(ParticleIdentifier.from(spec.location))
+
+        val particleContext = lookupParticleContextOrCreate(
+            context,
+            spec,
+            particle
+        )
+
+        for (handleSpec in spec.handles) {
+            val handle = createHandle(handleSpec.key, handleSpec.value, particle.handles)
+            particleContext.handles[handleSpec.key] = handle
+        }
+
+        return particleContext
+    }
+
+    fun lookupParticleContextOrCreate(
+        context: ArcHostContext,
+        spec: Plan.Particle,
+        particle: Particle
+    ) = context.particles[spec]?.copy(particle = particle) ?: ParticleContext(particle)
+
+    suspend fun performLifecycleForContext(context: ArcHostContext) {
+        for (particleContext in context.particles.values) {
+            performParticleLifecycle(particleContext)
+        }
+    }
+
+    suspend fun performParticleLifecycle(particleContext: ParticleContext) {
+        if (particleContext.particleState == ParticleState.Instantiated) {
+            particleContext.particle.onCreate()
+            particleContext.particleState = ParticleState.Created
+        }
+
+        // TODO: verify this won't be true if the host crashes
+        if (particleContext.particleState == ParticleState.Started) {
+            // particle already started
+            return
+        }
+
+        // If we reach here, particle is being restarted
+        if (particleContext.particleState == ParticleState.Stopped) {
+            particleContext.particleState = ParticleState.Created
+        }
+
+        // This is temporary until the BaseParticle PR lands and onStartup() API lands.
+        // We force sync() calls in lieu of onStartup() API for demos
+        if (particleContext.particleState == ParticleState.Created) {
+            for ((handleName, handle) in particleContext.handles) {
+                particleContext.run {
+                    particle.onHandleSync(handle, false)
+                }
+            }
+            particleContext.particleState = ParticleState.Started
+        }
+    }
+
+    suspend fun maybeRequestResurrection(context: ArcHostContext) {
+        if (context.arcState == ArcState.NeverStarted) {
+            // TODO: First time, request resurrection for context.
+        }
+    }
+
+    suspend fun maybeCancelResurrection(context: ArcHostContext) {
```

Comment:
KDoc please. This seems like an important function which should be well-documented for future engineers. Including when it might be used.  

Can this be private?